src/common/logging.rs

```diff
@@ -0,0 +1,272 @@
+use log::{error, info};
+use std::fs;
+use std::{
+    fs::copy,
+    path::{Path, PathBuf},
+};
+
+use crate::common::{path_append, ToError};
+use crate::{
+    common::{
+        debug,
+        defs::{BALENA_DATA_FSTYPE, NIX_NONE},
+        disk_util::DEF_BLOCK_SIZE,
+        error::Result,
+        loop_device::LoopDevice,
+    },
+    stage2::get_partition_infos,
+};
+
+use nix::{
+    mount::{mount, umount, MsFlags},
+    unistd::sync,
+};
+
+use self::stage2_config::Stage2Config;
+
+use super::{
+    defs::{BALENA_PART_MP, OLD_ROOT_MP},
+    reboot, stage2_config,
+};
+
+pub const LOG_TMPFS_DESTINATION: &str = "/tmp";
+pub const LOG_STAGE_1: &str = "stage1.log";
+pub const LOG_STAGE_2_INIT: &str = "stage2-init.log";
+pub const LOG_STAGE_2: &str = "stage2.log";
+pub const LOG_PRE_UNMOUNT_DATA_PART_DEST: &str = "/mnt/data/balenahup/takeover";
```

Comment:
This location should be configurable for non-balenaOS source distro. If we don't do this now, please document the items that must be updated for this goal. This feature is valuable for non-balenaOS sources, which likely are out in the field.

Reply:
the current implementation is scoped to balenaOS->balenaOS migration
We make assumptions about the filesystem and log to `/mnt/data/balenahup`

on balenaOS we know we can write there before and after flashing occurs during the takeover process. For other distros, we will need to test and come up with some alternative or perhaps the current mechanism will work.